maestro/src/agents/crewai_agent.py

```diff
@@ -2,116 +2,206 @@
 
 import importlib
 import asyncio
+from .agent import Agent as BeeAgent # Import BeeAgent first
+
+try:
+    # Only import crewai types if the library is present
+    from crewai import Agent as CrewAI_Agent, Crew, Task, Process
+    from crewai import LLM
+    enabled=True
+    CREWAI_IMPORT_ERROR = None
+except ImportError as e:
+    print("WARNING: Could not import crewai. CrewAI agent support will be disabled.")
+    enabled=False
+    CREWAI_IMPORT_ERROR = e # Store the original error
+    # Define dummy types to prevent NameErrors later if needed for type hints (though avoided below)
+    class CrewAI_Agent: pass
+    class Crew: pass
+    class Task: pass
+    class Process: pass
+    class LLM: pass
 
-from crewai import Agent, Crew, Task, Process, LLM
-from .agent import Agent as BeeAgent
 
 class CrewAIAgent(BeeAgent):
     """
     CrewAIAgent extends the Agent class to load and run a specific CrewAI agent.
+    Requires the 'crewai' library to be installed.
     """
-    def __init__(self, agent: dict) -> str:
+    def __init__(self, agent: dict) -> None:
         """
-        Initializes the workflow for the specified agent. 
+        Initializes the workflow for the specified agent.
         The executable code must be within $PYTHONPATH.
+
         Args:
-            agent_name (dict): Agent Configuration
+            agent (dict): Agent Configuration
+
         Raises:
-            Exception: If the agent cannot be loaded, an exception is raised with an error message.
+            RuntimeError: If the 'crewai' library is not installed or failed to import.
+            Exception: If the agent configuration is invalid.
         """
+        if not enabled:
```

Comment:
This check seems enough.  Is it OK to remove `_ensure_enabled` and the use of this function?

Reply:
Good spot - I'd iterated a few times. Cleaned up.

Long term it may be a useful approach for any optional dependencies -- it's what we ended up doing in beeai-framework.

I also refined the warning (though it is explicit, so will need updating in future).